how-to/customize-workspace/client/src/init-options.ts

```diff
@@ -0,0 +1,216 @@
+			listenerId = customParamIds[i];
+			break;
+		}
+	}
+	if (listenerId !== undefined) {
+		console.log(
+			`Init param has been passed and it has a matching listener (${listenerId}). Passing on init options to listeners`
+		);
+		const availableListeners = listeners.get(listenerId);
+		if (availableListeners !== undefined && availableListeners !== null) {
+			const subscriberIds = Array.from(availableListeners.keys());
+
+			for (let l = 0; l < subscriberIds.length; l++) {
+				const subscriberId = subscriberIds[l];
+				console.log(
+					`Init Options: Notifying subscriber with subscription Id: ${subscriberId} of request: ${listenerId}`
+				);
+				await availableListeners.get(subscriberId)(initOptions);
+			}
+		}
+	}
+}
+
+async function queryOnLaunch(userAppConfigArgs?: { [key: string]: string | boolean | number }) {
+	if (userAppConfigArgs !== undefined) {
+		console.log("Received during startup:", userAppConfigArgs);
+		if (userAppConfigArgs[actionParamName] !== undefined) {
+			await notifyActionListeners(userAppConfigArgs);
+		} else {
+			await notifyListeners(userAppConfigArgs);
+		}
+	}
+}
+
+async function queryWhileRunning(event: {
+	userAppConfigArgs?: { [key: string]: string | boolean | number };
+}) {
+	if (event?.userAppConfigArgs !== undefined) {
+		console.log("Received while platform is running:", event.userAppConfigArgs);
+		if (event.userAppConfigArgs[actionParamName] !== undefined) {
+			await notifyActionListeners(event.userAppConfigArgs);
+		} else {
+			await notifyListeners(event.userAppConfigArgs);
+		}
+	}
+}
+
+export async function init(options?: InitOptionsProviderOptions) {
+	if (initialized) {
+		return;
+	}
+	initialized = true;
+	if (Array.isArray(options?.modules)) {
+		for (let i = 0; i < options.modules.length; i++) {
+			const module = options.modules[i];
+			if (Array.isArray(module?.supportedActions) && module.supportedActions.length > 0) {
+				const isModuleLoaded = await loadInitOptionsModule(module.id, module.url, module.data);
+				if (isModuleLoaded) {
+					const loadedModule = modules[module.id];
+					if (loadedModule.init !== undefined) {
+						await loadedModule.init(module.data);
+					}
+					for (let a = 0; a < module.supportedActions.length; a++) {
+						const supportedAction = module.supportedActions[a];
+						registerActionListener(supportedAction, async (requestedAction: string, payload?: unknown) => {
+							const actionHandler = modules[module.id];
+							if (actionHandler !== undefined) {
+								console.log(`Action: ${requestedAction} being handled by module ${module.id}`);
+								await actionHandler.action(requestedAction, payload);
+							} else {
+								console.warn(
+									`Unable to retrieve module with id: ${module.id} to execute action: ${requestedAction}`
+								);
+							}
+						});
+					}
+				}
+			}
+		}
+	}
+	// eslint-disable-next-line @typescript-eslint/dot-notation
+	fin["desktop"].main(queryOnLaunch);
```

Comment:
Is it safer to use the App getinfo
```
const app = fin.Application.getCurrentSync();
const appInfo = await app.getInfo();
const customInitOptions = appInfo.initialOptions;
```

Reply:
Yeah I thought about that. It is a nicer api. And it should be populated by the time it executes

Reply:
updated.